src/app/lightning/get-balances.ts

```diff
@@ -23,6 +23,50 @@ export const getTotalBalance = async (): Promise<Satoshis | ApplicationError> =>
   return sumBalances(balances)
 }
 
+export const getInboundBalance = async (): Promise<Satoshis | ApplicationError> => {
+  const offChainService = LndService()
+  if (offChainService instanceof Error) return offChainService
+  const offChainChannelBalances = await offChainService.getInboundOutboundBalance()
+  if (offChainChannelBalances instanceof Error) return offChainChannelBalances
+
+  const inboundOutboundBalance = await Promise.all(
+    offChainService
+      .listActivePubkeys()
+      .map((pubkey) => offChainService.getInboundOutboundBalance(pubkey)),
+  )
+
+  const inbound: Satoshis[] = []
+  inboundOutboundBalance.forEach((balance) => {
+    if ("inbound" in balance) {
+      inbound.push(balance.inbound)
+    }
+  })
+
+  return sumBalances(inbound)
+}
+
+export const getOutboundBalance = async (): Promise<Satoshis | ApplicationError> => {
+  const offChainService = LndService()
+  if (offChainService instanceof Error) return offChainService
+  const offChainChannelBalances = await offChainService.getInboundOutboundBalance()
+  if (offChainChannelBalances instanceof Error) return offChainChannelBalances
+
+  const inboundOutboundBalance = await Promise.all(
+    offChainService
+      .listActivePubkeys()
+      .map((pubkey) => offChainService.getInboundOutboundBalance(pubkey)),
+  )
+
+  const outbound: Satoshis[] = []
+  inboundOutboundBalance.forEach((balance) => {
+    if ("outbound" in balance) {
+      outbound.push(balance.outbound)
+    }
+  })
```

Comment:
you can replace this by a map 
```suggestion
  const outbound = inboundOutboundBalance.map(b => b.outbound)
```

Reply:
same here

Reply:
the suggestion was causing a type error: https://github.com/GaloyMoney/galoy/pull/2057/commits/2eb0640a00755b3f41c8ff24e81d40e73e3ae121 so reverted to the original solution

Reply:
```
const outbound = inboundOutboundBalance.map((balance) => {
    if (balance instanceof Error) return toSats(0) 
    return balance.outbound
  })
 ```